scripts/check-workflow-determinism.test.sh

```
@@ -0,0 +1,452 @@
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+SCRIPT_UNDER_TEST="$SCRIPT_DIR/check-workflow-determinism.sh"
+PASS=0
+FAIL=0
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
```

Shellcheck (0.11.0)

[warning] YELLOW appears unused. Verify use (or export if used externally).

(SC2034)